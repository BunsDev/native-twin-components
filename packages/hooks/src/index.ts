export { useBlurredBackgroundStyles } from './hooks/useBlurredBackgroundStyles';
export { useCalendarSet } from './hooks/useCalendarSet';
export { useCallbackRef } from './hooks/useCallbackRef';
export { useClickAway } from './hooks/useClickAway';
export { useIsDarkMode } from './hooks/useIsDarkMode';
export { useIsMobileWeb } from './hooks/useIsMobileWeb';
export { useIsSmallScreen } from './hooks/useIsSmallScreen';
export { useLayout } from './hooks/useLayout';
export { useLockBodyScroll } from './hooks/useLockBodyScroll';
export { useOnFocus } from './hooks/useOnFocus';
export { useOnHover } from './hooks/useOnHover';
export { useOnPress } from './hooks/useOnPress';
export { usePlatformResize } from './hooks/usePlatformResize';
export { useUpdateEffect } from './hooks/useUpdateEffect';
export { useWebClientRect } from './hooks/useWebClientRect';
export type { PlatformRect } from './hooks/useWebClientRect';
export { useWebScroll } from './hooks/useWebScroll';
export { getScrollParent } from './utils/get-scroll-parent';
