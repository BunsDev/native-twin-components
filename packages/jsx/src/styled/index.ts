export {
  createStylableComponent,
  withMappedProps,
  twinComponent,
  stylizedComponents,
} from './native';
