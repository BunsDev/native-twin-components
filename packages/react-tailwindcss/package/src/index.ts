import { styled, useIsDarkMode, useStyledComponent } from '@react-universal/core';

export { styled, useIsDarkMode, useStyledComponent };
