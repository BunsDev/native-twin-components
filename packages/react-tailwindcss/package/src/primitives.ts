export * from '@react-universal/primitives';
