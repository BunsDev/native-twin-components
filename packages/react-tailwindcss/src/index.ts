export * from '@react-universal/core';
