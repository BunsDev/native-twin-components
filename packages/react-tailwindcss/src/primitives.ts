export * from '@universal-labs/primitives';
