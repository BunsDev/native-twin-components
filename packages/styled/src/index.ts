export { useIsDarkMode } from './hooks/useIsDarkMode';
export { styled } from './styled';
export type { ForwardedStyledComponent } from './styled/StyledComponent';
export { useBuildStyledComponent } from './hooks/useBuildStyledComponent';
