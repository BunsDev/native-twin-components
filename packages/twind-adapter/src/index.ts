export {
  tx,
  tw,
  stringify,
  setTailwindConfig,
  parse,
  escape,
  inline,
  shortcut,
} from './client';

export { transformClassNames } from './utils/transformClassNames';
export { normalizeClassNameString } from './utils/helpers';
