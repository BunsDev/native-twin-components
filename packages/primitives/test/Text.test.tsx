import renderer from 'react-test-renderer';
import { describe, expect, it } from 'vitest';
import { H1 } from '../src';

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

describe('@universal-labs/primitives', () => {
  it('H1 render', () => {
    const component = renderer.create(<H1 className='text-sm'>Test H1</H1>);
    let tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
