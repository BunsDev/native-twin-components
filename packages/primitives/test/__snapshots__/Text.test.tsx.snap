// Vitest Snapshot v1

exports[`@universal-labs/primitives > H1 render 1`] = `
<h1
  aria-level={1}
  className="css-text-1rynq56 flex text-4xl font-bold text-sm"
  dir="auto"
  role="heading"
>
  Test H1
</h1>
`;
